crates/fuel-core/src/query/message.rs

```diff
@@ -185,6 +186,8 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
         None => return Ok(None),
     };
 
+    let message_id = compute_message_id(&sender, &recipient, &nonce, amount, &data);
```

Comment:
If I understand correctly, this is kinda the meat of the change: Rather than using the message id directly, we pass in the nonce. Together, these pieces of information allow us to reconstruct the message id.

Reply:
Yep.  Maybe I could have reworded the contents of the issue.

We were just asking for more information than needed.